runtime-modules/content/src/benchmarks/benchmarking.rs

```diff
@@ -1797,6 +1792,642 @@ benchmarks! {
+        let origin = RawOrigin::Signed(curator_account_id.clone());
+        let actor = ContentActor::Curator(group_id, curator_id);
+        let (_, video_state_bloat_bond, data_object_state_bloat_bond, _) = setup_bloat_bonds::<T>()?;
+        let video_id = Pallet::<T>::next_video_id();
+        set_nft_limits_helper::<T>(channel_id);
+        Pallet::<T>::create_video(origin.clone().into(), actor.clone(), channel_id, VideoCreationParameters::<T> {
+            expected_video_state_bloat_bond: video_state_bloat_bond,
+            expected_data_object_state_bloat_bond: data_object_state_bloat_bond,
+            assets: None,
+            auto_issue_nft: None,
+            meta: None,
+            storage_buckets_num_witness: storage_buckets_num_witness::<T>(channel_id)?,
+        })?;
+
+        let params = worst_case_nft_issuance_params_helper::<T>(w,b);
+    }: _ (origin, actor, video_id, params)
+        verify {
+            assert!(Pallet::<T>::video_by_id(video_id).nft_status.is_some());
+        }
+
+    // WORST CASE SCENARIO:
+    // STATE COMPLEXITY
+    // - curator owned channel
+    // - curator number is max
+    // - curator has max number of permissions
+    // - channel has all features paused except necessary ones
+    // - channel has max assets
+    // - NFT owner == channel owner
+    // INPUT COMPLEXITY
+    destroy_nft {
+        let (channel_id, group_id, lead_account_id, curator_id, curator_account_id) =
+            setup_worst_case_scenario_curator_channel_all_max::<T>(false)?;
+        let origin = RawOrigin::Signed(curator_account_id.clone());
+        let actor = ContentActor::Curator(group_id, curator_id);
+        let video_id = setup_video_with_idle_nft::<T>(curator_account_id.clone(), actor, channel_id)?;
+        let origin = RawOrigin::Signed(curator_account_id);
+    }: _ (origin, actor, video_id)
+        verify {
+            assert!(Pallet::<T>::video_by_id(video_id).nft_status.is_none());
+        }
+
+    // WORST CASE SCENARIO:
+    // STATE COMPLEXITY
+    // - curator owned channel
+    // - curator number is max
+    // - curator has max number of permissions
+    // - channel has all features paused except necessary ones
+    // - channel has max assets
+    // - NFT owner == channel owner
+    // INPUT COMPLEXITY
+    sling_nft_back {
+        let (channel_id, group_id, lead_account_id, curator_id, curator_account_id) =
+            setup_worst_case_scenario_curator_channel_all_max::<T>(false)?;
+        let origin = RawOrigin::Signed(curator_account_id.clone());
+        let actor = ContentActor::Curator(group_id, curator_id);
+        let video_id = setup_video_with_idle_nft::<T>(curator_account_id.clone(), actor, channel_id)?;
+        let origin = RawOrigin::Signed(curator_account_id);
+    }: _ (origin, video_id, actor)
+        verify {
+            assert!(Pallet::<T>::video_by_id(video_id).nft_status.unwrap().owner == NftOwner::ChannelOwner);
```

Comment:
Adding to the previous comment:
It looks like currently triggering `sling_nft_back` with `NftOwner::Channel` is indeed the worst case scenario, because of `ensure_actor_authorized_to_manage_nft`, but it's also a scenario that makes no sense in practice, because it results in a no-op (yet still consumes resources). I think it'd be best to return `Err` in `sling_nft_back` if the owner is already a channel.